### PR TITLE
Improve Java Hello World

### LINKED ISSUE #970: Pin Java examples base docker image
Currently the Java example uses `openjdk:8-jre-alpine`. However, some JVM version will support container support out of the box (8u191 and on), while older version of may require additional flags.

If a wrong version of openjdk:8-jre-alpine was used (because it was cached), then the app may see OOMKilled, since it won't respect cgroup memory limit.

Moreover, openjdk8 official image don't have the latest updates (8u202, for example).

## Recommendation
Change base Java image to AdoptOpenJDK & pin version: `adoptopenjdk/openjdk8:jdk8u202-b08-alpine-slim`

Or, since AdoptOpenJDK images always configures the container support flags, any image versions will work: `adoptopenjdk/openjdk8:alpine-slim`

### PR DESCRIPTION
**Pull request by saturnism**
Fixes #970

## Proposed Changes

- Use AdoptOpenJDK for newest version of JDK that has proper container support.
- Upgraded Spring Boot version.

### CODE REVIEW

File: docs/serving/samples/hello-world/helloworld-java/pom.xml [hunk 1]
```diff
@@ -14,7 +14,7 @@
 	<parent>
 		<groupId>org.springframework.boot</groupId>
 		<artifactId>spring-boot-starter-parent</artifactId>
-		<version>2.0.3.RELEASE</version>
+		<version>2.1.3.RELEASE</version>
```
Comment (gyliu513) [inline docs/serving/samples/hello-world/helloworld-java/pom.xml]: Why upgrade to 2.1.3 directly? What is the versioning guidance for this?

File: docs/serving/samples/hello-world/helloworld-java/.dockerignore [hunk 2]
```diff
@@ -0,0 +1 @@
+target/
```
Comment (gyliu513) [inline docs/serving/samples/hello-world/helloworld-java/.dockerignore]: Does target will be generated during docker build?

File: docs/serving/samples/hello-world/helloworld-java/pom.xml (hunk 1 above)
Comment (saturnism) [reply docs/serving/samples/hello-world/helloworld-java/pom.xml]: it's the latest GA version.

File: docs/serving/samples/hello-world/helloworld-java/.dockerignore (hunk 2 above)
Comment (saturnism) [reply docs/serving/samples/hello-world/helloworld-java/.dockerignore]: this ignores locally generated target/ dir, so we avoid copying unnecessary files to docker daemon during build

**Comment by saturnism** [comment]
not sure why the test failed; is it flaky test? this pr shouldn't affect the tests.

**Comment by RichieEscarez** [comment]
> not sure why the test failed; is it flaky test? this pr shouldn't affect the tests.

the integration tests for samples have only been running for a few weeks now so it might be possible that either the test is failing on something outside of your set of changes or the test itself is broken?

it might be the latter because its not finding the build artifacts:
`W0409 20:33:58.432] Remote dir gs://knative-prow/pr-logs/pull/knative_docs/1111/pull-knative-docs-integration-tests/1115714356525404160/artifacts not exist yet`

@chaodaiG can you please help us determine how to get past the build failure?

**Comment by chaodaiG** [comment]
@RichieEscarez , there was an issue for all knative testing on Tuesday, it was resolved. I just requeued the test, it should pass now

**Comment by RichieEscarez** [comment]
> @RichieEscarez , there was an issue for all knative testing on Tuesday, it was resolved. I just requeued the test, it should pass now

Thanks!

**Pull request merged**